### PR TITLE
Modify JS version of Presence tutorial

### PR DESCRIPTION
**Pull request by marklewin**
Modify JS version of the Presence tutorial to use Promise interface and ES6 syntax.

### CODE REVIEW

File: example.html [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
-      channel.presence.subscribe(function(presenceMsg) {
-        console.log('Received a ' + presenceMsg.action + ' from ' + presenceMsg.clientId);
-        channel.presence.get(function(err, members) {
-          if(err) { return console.log("Error fetching presence data: " + err); }
-          document.getElementById('presence-set').innerHTML = members.map(function(member) {
-            return '<li>' + member.clientId + '</li>';
-          }).join();
+      // Subscribe to the presence set to receive updates
+      await channel.presence.subscribe((presenceMessage) => {
+        console.log(
```
Comment (fliptopbox) [inline example.html]: please use console.log argument list, instead of String concatenation, and reduce to a single line.

File: example.html [hunk 2]
```diff
@@ -1,43 +1,73 @@
...
-    alert("You need to add your API key"); /* REMOVE THIS */
-    var realtime = new Ably.Realtime({
-      key: "INSERT-YOUR-API-KEY-HERE", /* ADD YOUR API KEY HERE */
-      clientId: "my-client-id", /* This is who you will appear as in the presence set */
-      closeOnUnload: true // See https://support.ably.io/solution/articles/3000059875-why-don-t-presence-members-leave-as-soon-as-i-close-a-tab-
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY"; // Replace with your API key
+    const randomId =
+      Math.random().toString(36).substring(2, 10) +
+      Math.random().toString(36).substring(2, 10);
```
Comment (fliptopbox) [inline example.html]: a simple alternative 

```suggestion
    const randomId = Math.random().toString(36).slice(-10);
```

File: example.html [hunk 3]
```diff
@@ -1,43 +1,73 @@
...
+        // Update the list of channel members when the presence set changes
+        channel.presence.get((err, members) => {
+          if (err) {
+            return console.error(`Error retrieving presence data: ${err}`);
+          }
+          document.getElementById("presence-set").innerHTML = members
+            .map((member) => {
+              return "<li>" + member.clientId + "</li>";
+            })
+            .join("");
```
Comment (fliptopbox) [inline example.html]: Please use String literals so the method reads easier.

```suggestion
          document.getElementById("#presence-set").innerHTML = members
            .map(({clientid}) => `<li>${clientId}</li>`)
            .join("");
```

**Review by fliptopbox — CHANGES_REQUESTED** [review]
how close do we want our `node` and `javascript` code style to be?
All `nodejs` versions > 12 can use ES6 so we could use almost identical source.

File: example.html (hunk 1 above)
Comment (marklewin) [reply example.html]: 👍

File: example.html (hunk 3 above)
Comment (marklewin) [reply example.html]: 👍

File: example.html (hunk 2 above)
Comment (marklewin) [reply example.html]: Nice

**Comment by marklewin** [comment]
> how close do we want our `node` and `javascript` code style to be? All `nodejs` versions > 12 can use ES6 so we could use almost identical source.

Yes! The plan in the tutorial is to do just that. Only the header stuff (require() vs <script>) and DOM manipulation (in the JS version) is different.

**Review by fliptopbox — CHANGES_REQUESTED** [review]
please replace ably.io URLs and old logo in README

**Review by fliptopbox — APPROVED** [review]

**Pull request merged**